ThirdPartyModules/Magento/GiftCardAccount.php

```diff
@@ -196,4 +196,127 @@ function boltMagentoGiftInvalidateCart() {
+                }
+            }
+
+            if ($parentQuote->getGiftCardsAmountUsed() == 0) {
+                try {
+                    // on subsequest validation calls from Bolt checkout
+                    // try removing the gift card before adding it
+                    $giftCard->removeFromCart(true, $parentQuote);
+                } catch (\Exception $e) {
+                    // gift card not added yet
+                } finally {
+                    $giftCard->addToCart(true, $parentQuote);
+                }
+            }
+
+            // Send the whole GiftCard Amount.
+            $giftAmount = $parentQuote->getGiftCardsAmount();
+            $currencyCode = $parentQuote->getQuoteCurrencyCode();
+
+            return [
+                'status'          => 'success',
+                'discount_code'   => $code,
+                'discount_amount' => abs(
+                    CurrencyUtils::toMinor($giftAmount, $currencyCode)
+                ),
+                'description'     => __('Gift Card (%1)', $code),
+                'discount_type'   => 'fixed_amount',
+            ];
+        } catch (\Exception $e) {
+            return [
+                'status'        => 'failure',
+                'error_message' => $e->getMessage(),
+            ];
+        }
+    }
+    
+    /**
+     * @param bool                                           $result
+     * @param string                                         $couponCode
+     * @param \Magento\GiftCardAccount\Model\Giftcardaccount $giftCard
+     * @param Quote                                          $quote
+     * @return bool
+     */
+    public function filterApplyingGiftCardCode(
+        $result,
+        $couponCode,
+        $giftCard,
+        $quote
+    ) {
+        if ($result || !$giftCard instanceof \Magento\GiftCardAccount\Model\Giftcardaccount) {
+            return $result;
+        }
+        
+        try {
+            // on subsequest validation calls from Bolt checkout
+            // try removing the gift card before adding it
+            $giftCard->removeFromCart(true, $quote);
+        } catch (\Exception $e) {
+            // gift card not added yet
+        } finally {
+            $giftCard->addToCart(true, $quote);
+            return true;
+        }
+    }
+    
+    /**
+     * @param bool                                          $result
+     * @param Magento\GiftCardAccount\Model\Giftcardaccount $giftCard
+     * @param Quote                                         $quote
+     *
+     * @return bool
+     */
+    public function filterRemovingGiftCardCode(
+        $result,
+        $giftCard,
+        $quote
+    ) {
+        if ($result || !$giftCard instanceof \Magento\GiftCardAccount\Model\Giftcardaccount) {
+            return $result;
+        }
+        
+        try {
+            // on subsequest validation calls from Bolt checkout
+            // try removing the gift card before adding it
```

Comment:
This comment is redundant, please remove or change it.

Reply:
Removed